dht-node/src/dht_node/index.ts

```diff
@@ -190,20 +198,69 @@ async function writeFederatedHomeCommunityEnries(pubKey: any): Promise<Community
   try {
     // first remove privious existing homeCommunity entries
     DbFederatedCommunity.createQueryBuilder().delete().where({ foreign: false }).execute()
-
-    homeApiVersions.forEach(async function (homeApi) {
-      const homeCom = new DbFederatedCommunity()
+    for (let i = 0; i < homeApiVersions.length; i++) {
+      const homeCom = DbFederatedCommunity.create()
       homeCom.foreign = false
-      homeCom.apiVersion = homeApi.api
-      homeCom.endPoint = homeApi.url
-      homeCom.publicKey = pubKey.toString('hex')
-
-      // this will NOT update the updatedAt column, to distingue between a normal update and the last announcement
+      homeCom.apiVersion = homeApiVersions[i].api
+      homeCom.endPoint = homeApiVersions[i].url
+      homeCom.publicKey = Buffer.from(pubKey)
       await DbFederatedCommunity.insert(homeCom)
-      logger.info(`federation home-community inserted successfully: ${JSON.stringify(homeCom)}`)
-    })
+      logger.info(
+        `federation home-community inserted successfully: ${JSON.stringify(homeApiVersions[i])}`,
+      )
+    }
   } catch (err) {
-    throw new Error(`Federation: Error writing HomeCommunity-Entries: ${err}`)
+    throw new Error(`Federation: Error writing federated HomeCommunity-Entries: ${err}`)
   }
   return homeApiVersions
 }
+
+export async function writeHomeCommunityEntry(pubKey: string): Promise<void> {
+  try {
+    // check for existing homeCommunity entry
+    let homeCom = await DbCommunity.findOne({
+      foreign: false,
+      publicKey: Buffer.from(pubKey),
+    })
+    if (!homeCom) {
+      // check if a homecommunity with a different publicKey still exists
+      homeCom = await DbCommunity.findOne({ foreign: false })
```

Comment:
on my opinion it is a difference to update an existing homeCommunity-entry with the correct publicKey and to keep the id and uuid of this entry.
the second is to check if a homeCommunity-entry exists, but with a different publicKey (in case no seed axists) which will update all necessary attributes